Add update 10 blog (#269)
ballerina-platform/blog: _posts/2024-08-22-announcing-ballerina-2201.10.0-swan-lake-update-10.md

---
layout: blog-post
title: Announcing Ballerina Swan Lake Update 10 (2201.10.0)

> author: Ballerina Team

Reviewer: Lets change it to Shafreen?

published-date: 22 Aug 2024
status: Published
socialmediaimage: ballerina-generic-social-media-image-2023.png
permalink: /posts/2024-08-22-announcing-ballerina-2201.10.0-swan-lake-update-10/
---

<style>.cBlogContent p{white-space: break-spaces !important;}</style>

> We are excited to announce that Ballerina Swan Lake has just released its tenth update release, Ballerina 2201.10.0 (Swan Lake Update 10). This update brings enhancements across various areas, with key highlights including a revamped OpenAPI tool, a series of new connectors, and innovative AI features.

Reviewer: ```suggestion
We are excited to announce that Ballerina Swan Lake has just released its tenth update release, Ballerina 2201.10.0 (Swan Lake Update 10). This update brings enhancements across various areas, with key highlights including a revamped OpenAPI tool, a series of new connectors, and innovative AI features.
```
```suggestion
We are excited to announce that Ballerina has just released its tenth update release, Ballerina Swan Lake Update 10(2201.10.0). This update brings enhancements across various areas, with key highlights including a revamped OpenAPI tool, a series of new connectors, and innovative AI features.
```

> We are excited to announce that Ballerina Swan Lake has just released its tenth update release, Ballerina 2201.10.0 (Swan Lake Update 10). This update brings enhancements across various areas, with key highlights including a revamped OpenAPI tool, a series of new connectors, and innovative AI features.

Reviewer 2: As suggested on the doc,

"We are excited to announce the tenth update release of Ballerina Swan Lake"

or has to be something like 

"We are excited to announce that the Ballerina team has just released the tenth update ..."?

or

"The Ballerina team is excited to announce the tenth update ..."

Since we/the team are releasing the update, not Ballerina Swan Lake.

## OpenAPI Tool

A major highlight of Ballerina Swan Lake Update 10 is the introduction of extensive enhancements to the OpenAPI tool. 

> A contract-first approach is introduced as a standout key feature using the OpenAPI tool and the HTTP package. By generating a service contract type directly from the OpenAPI specification, you can design services that are contract-compliant from the outset. In addition, centralizing metadata within the service contract type results in a cleaner, more maintainable codebase. 

Reviewer 2: ```suggestion

A contract-first approach is introduced using the OpenAPI tool and the HTTP package. By generating a service contract type directly from the OpenAPI specification, you can design services that are contract-compliant from the outset. In addition, centralizing metadata within the service contract type results in a cleaner, more maintainable code base. 
```

<img alt="Swagger UI" src="/images/swagger_ui.png">

> Another important feature is the addition of example annotations on Ballerina types, resource parameters, and record fields, which are now directly rendered into the OpenAPI example schema. This provides developers with clear, contextually relevant examples, making APIs easier to understand and implement. For more information, check this [link](https://ballerina.io/learn/openapi-tool/#export-with-given-examples-information). 

Reviewer: Instead of saying `check this link` give meaning full link texts as per the writing practices. Refer to previous blogs and guideline [1] Check all the places.

[1] https://learn.microsoft.com/en-us/style-guide/urls-web-addresses#grammar-and-formatting

> Another important feature is the addition of example annotations on Ballerina types, resource parameters, and record fields, which are now directly rendered into the OpenAPI example schema. This provides developers with clear, contextually relevant examples, making APIs easier to understand and implement. For more information, check this [link](https://ballerina.io/learn/openapi-tool/#export-with-given-examples-information). 

Reviewer 2: ```suggestion
Another important feature is the support for example annotations on Ballerina types, resource parameters, and record fields, which are now directly rendered into the OpenAPI example schema. This provides developers with clear, contextually relevant examples, making APIs easier to understand and implement. For more information, check [Export with examples](https://ballerina.io/learn/openapi-tool/#export-with-given-examples-information). 
```

> To further streamline development, the update introduces mock client generation with examples directly from the OpenAPI specification. Picture a developer tasked with integrating a new API: you can now effortlessly generate mock clients with examples that you documented in the OAS, test and validate integrations, and then seamlessly replace the mock client with the actual one when ready for production. For more information, check this [link](https://ballerina.io/learn/openapi-tool/#generate-mock-client-using-included-example-in-oas).

Reviewer: ```suggestion
To further streamline development, this update introduces mock client generation with examples directly from the OpenAPI specification. Picture a developer tasked with integrating a new API: you can now effortlessly generate mock clients with examples that you documented in the OAS, test and validate integrations, and then seamlessly replace the mock client with the actual one when ready for production. For more information, check this [link](https://ballerina.io/learn/openapi-tool/#generate-mock-client-using-included-example-in-oas).
```

The OpenAPI package allows developers to automatically generate and embed the OpenAPI specification for a Ballerina service via a resource. This has been improved by exposing another resource to view the HTML-rendered version of the OpenAPI specification. This feature allows you to introspect the service through a more intuitive and readable format. For more information, see [OpenAPI Specification resources](https://ballerina.io/spec/http/#236-openapi-specification-resources)

<img alt="Service contract type" src="/images/service_contract_type.png">

> ## New Connectors

Reviewer: We better either link each individual connector in central or at one generic line at the end of this section asking the users to visit central and find these connectors. For any new ballerina user, it will be hard to find where these connectors are without any reference.

Reviewer 3: @NipunaRanasinghe

Reviewer 4: @anupama-pathirage yes, agreed. Updated each connector reference with its link to the central

> In this release, we’ve made substantial improvements to the functionality and usability of our connectors. Here are the key highlights:

Reviewer 2: ```suggestion
In this release, we’ve made substantial improvements to the functionality and usability of some connectors. Here are the key highlights:
```

> As part of the connector revamp initiative, several connectors have been significantly updated. The newly released `dayforce` and `discord` packages now support interactions with their latest REST API versions. Additionally, `asb`, `slack`, `stripe`, and `twitter` (X) connectors have received major updates, featuring significant API and functionality changes, along with enhanced documentation, making it easier to develop complex applications.

Reviewer: Why `dayforce` and `discord` have referred as packages while the `asb`, `slack` etc have referred as connectors?

Reviewer 2: Also not consistent with the title and first sentence.

> As part of the connector revamp initiative, several connectors have been significantly updated. The newly released `dayforce` and `discord` packages now support interactions with their latest REST API versions. Additionally, `asb`, `slack`, `stripe`, and `twitter` (X) connectors have received major updates, featuring significant API and functionality changes, along with enhanced documentation, making it easier to develop complex applications.

Reviewer 2: Don't these have to be indented? Have we checked how all of these get rendered?

Reviewer 3: Added bullets

> As part of the connector revamp initiative, several connectors have been significantly updated. The newly released `dayforce` and `discord` packages now support interactions with their latest REST API versions. Additionally, `asb`, `slack`, `stripe`, and `twitter` (X) connectors have received major updates, featuring significant API and functionality changes, along with enhanced documentation, making it easier to develop complex applications.

Reviewer 2: significantly updated or updated significantly?

> As part of the connector revamp initiative, several connectors have been significantly updated. The newly released `dayforce` and `discord` packages now support interactions with their latest REST API versions. Additionally, `asb`, `slack`, `stripe`, and `twitter` (X) connectors have received major updates, featuring significant API and functionality changes, along with enhanced documentation, making it easier to develop complex applications.

Reviewer 2: interactions can be just interaction?

> As part of the connector revamp initiative, several connectors have been significantly updated. The newly released `dayforce` and `discord` packages now support interactions with their latest REST API versions. Additionally, `asb`, `slack`, `stripe`, and `twitter` (X) connectors have received major updates, featuring significant API and functionality changes, along with enhanced documentation, making it easier to develop complex applications.

Reviewer 2: Additionally, `asb` -> Additionally, the `asb`

received -> undergone

For SAP integrations, the `sap` package has been upgraded to support the invocation of any OData endpoint in S/4HANA and now offers automatic CSRF token authentication, enhancing both security and usability. Similarly, the `sap.jco` package now supports the invocation of RFC modules with better type binding and allows the transmission and reception of iDocs. Additionally, the `sap.s4hana.sales` package provides a high-level interface for Sales and Distribution (SD) APIs in S/4HANA, enabling streamlined sales order management, delivery processing, and billing operations.

The Salesforce package now includes a listener with streaming API support for real-time data synchronization and event-driven workflows. Additionally, the introduction of the `salesforce.types` submodule enables the use of base types, and the reorganization of Salesforce clients enhances usability.

## AI features 

> Ballerina Swan Lake Update 10 contains a new feature for Ballerina Copilot and significant improvements to the existing AI data mapper as well. These experimental features will greatly enhance developer productivity and streamline the integration development lifecycle. 

Reviewer: `Ballerina Swan Lake Update 10 contains a new feature for Ballerina Copilot` - This part is not clear as it is not explaining what hs this new feature for Ballerina copilot. It is not the two points discussed below right? 

And IINM, copilot is not released with update 10. so is it correct to say this?

> - Test Generator 

Reviewer: We need to link any /learn page content for these. If that is not there yet, we need to add those with all new features.
